tools/drsnoop.py

```diff
@@ -48,22 +52,52 @@
 parser.add_argument("-n", "--name",
                     type=ArgString,
                     help="only print process names containing this name")
+parser.add_argument("-v", "--verbose", action="store_true",
+                    help="show system memory state")
 parser.add_argument("--ebpf", action="store_true",
                     help=argparse.SUPPRESS)
 args = parser.parse_args()
 debug = 0
 if args.duration:
     args.duration = timedelta(seconds=int(args.duration))
 
+
+# vm_stat
+
+vm_stat_addr = ''
+with open(kallsyms) as syms:
+    for line in syms:
+        (addr, size, name) = line.rstrip().split(" ", 2)
+        name = name.split("\t")[0]
+        if name == "vm_stat":
+            vm_stat_addr = "0x" + addr
+            break
+        if name == "vm_zone_stat":
+            vm_stat_addr = "0x" + addr
+            break
+    if vm_stat_addr == '':
+        print("ERROR: no vm_stat or vm_zone_stat in /proc/kallsyms. Exiting.")
+        print("HINT: the kernel should be built with CONFIG_KALLSYMS_ALL.")
+        exit()
+
+NR_FREE_PAGES = 0
+
+PAGE_SHIFT = 12    # defined in page_types.h
```

Comment:
different systems may have differeent PAGE_SHIFT. Maybe you need to get from sysconf and replace it with proper values?

Reply:
Agree! It seems no way to get `PAGE_SHIFT` directly,  maybe we can get `PAGE_SIZE` to cal `PAGE_SHIFT`. I'll make sure, and fix this.